app/controllers/director/kids_controller.rb

```diff
@@ -10,26 +10,22 @@ def new
 
   def create
     @kid = Kid.new(kid_params)
-
-    @parent = Parent.where("
-      user_id IN (
-        SELECT id FROM users
-        WHERE email = ?
-      )",
-      params[:parent_email]
-    )
-    @kid.parents << @parent
+    current_date = Date.today
+    age = ((current_date - @kid.birthdate).to_i)/365
+    @medical_info = MedicalInfo.new(kid_medical_params)
+    @parent = Parent.where("user_id = ?", User.find_by(email: params[:parent_email]).id)
+    @group = Group.where("min_age <= ? AND max_age >= ?", age, age)
 
     if @kid.save
-      @medical_info = MedicalInfo.new(kid_medical_params)
+      puts "KID SAVED"
       @medical_info.kid_id = @kid.id
-      if @medical_info.save
-        redirect_to director_kid_path(:id => @kid.id)
-      else
-        @kid.delete
-        flash[:alert] = @kid_medical.errors.full_messages
-        redirect_to new_director_kid_path
-      end
+      @medical_info.save
+      puts "MEDICAL INFO SAVED"
+      @kid.parents << @parent
+      puts "PARENTS SAVED"
+      @kid.groups << @group
+      puts "GROUPS SAVED"
+      redirect_to director_kid_path(:id => @kid.id)
```

Comment:
There are no error checks in any of this. what if the medicare wasn't put into the med info? what if the group doesn't exist? what if we put in the wrong email and nothing is returned? the kid will be saved, but these checks are not in place